daemon/src/identify/dialer.rs

```diff
@@ -76,13 +76,17 @@ impl Actor {
         let daemon_version = peer_info.daemon_version.clone();
         let environment = peer_info.environment;
 
-        NUM_LIBP2P_CONNECTIONS_GAUGE
-            .with(&HashMap::from([
-                (WIRE_VERSION_LABEL, wire_version.as_str()),
-                (DAEMON_VERSION_LABEL, daemon_version.as_str()),
-                (ENVIRONMENT_LABEL, environment.to_string().as_str()),
-            ]))
-            .inc();
+        // Only increment if we don't know the peer info already because sometimes we are not
+        // notified about ConnectionDropped
+        if self.peer_infos.get(&peer_id).is_none() {
```

Comment:
🙃 You could do the same thing a few lines below when inserting into the map: 
 
From rust doc: 
> Inserts a key-value pair into the map.
If the map did not have this key present, None is returned.
If the map did have this key present, the value is updated, and the old value is returned. 

e.g. 
```
if let None = self.peer_infos.insert(peer_id, peer_info.clone()) { 
... 
}
```

Reply:
maybe even `if (...).is_none` {
} 

no need to bind it with `let`

Reply:
I like the proposal, it's cleaner. Done!

Reply:
> 

true, could just go for `is_none` here

Reply:
clippy agrees as well 🤪